web/src/app/shared/components/type-ahead/type-ahead.component.ts

```diff
@@ -95,6 +97,10 @@ export class TypeAheadComponent implements AfterViewInit, ControlValueAccessor {
     }
   }
 
+  changeValue() {
+    this.initializeDataSource();
+  }
```

Comment:
Why don't you call initializeDataSource() instead of changeValue()? I think changeValue() is unnecessary.